drivers/SmartThings/zwave-garage-door-opener/src/ecolink-zw-gdo/init.lua

```diff
@@ -0,0 +1,458 @@
+
+  device:send(Configuration:BulkSetV2({
+                                        parameter_offset = 1, 
+                                        size = 2, 
+                                        handshake = false, 
+                                        default = false,
+                                        parameters = updated_params
+                                      }))
+end
+
+--- Handle Device Needs Configuring Event
+---
+--- @param driver st.zwave.Driver
+--- @param device st.zwave.Device
+local function do_configure(driver, device)
+  log.info_with({hub_logs=true}, "do_configure")
+  configure_device_with_updated_config(device)
+end
+
+local function device_preferences_updated(driver, device, event, args)
+  -- Did my preference value change
+  log.info_with({hub_logs=true}, 
+                string.format("device_preferences_updated> device:%s event:%s args:%s", 
+                tostring(device), tostring(event), tostring(args)))
+  configure_device_with_updated_config(device)
+end
+
+--- Notification Report Handler
+---
+--- @param driver st.zwave.Driver
+--- @param device st.zwave.Device
+--- @param cmd st.zwave.CommandClass.Notification.Report
+local function notification_report_handler(driver, device, cmd)
+  local notificationType = cmd.args.notification_type
+  local notificationEvent = cmd.args.event
+  local barrier_event = nil
+  local contact_event = nil
+
+  log.info_with({hub_logs=true}, 
+                string.format("notification_report_handler> Type:%d Event:%d", 
+                notificationType, notificationEvent) )
+
+  if ( 0 == notificationEvent ) then
+    -- Clear Notifications
+    -- First byte of the parameters is the notification being cleared
+    -- so reuse notificationEvent variable as the event being cleared
+    if (0 ~= string.len(cmd.args.event_parameter)) then
+      notificationEvent = string.byte(cmd.args.event_parameter)
+      log.info_with({hub_logs=true}, string.format("NOTIFICATION CLEARED EVENT"))
+
+    end
+
+    if (notificationType == Notification.notification_type.SYSTEM) then
+
+      if (notificationEvent == Notification.event.system.TAMPERING_PRODUCT_COVER_REMOVED) then
+        log.info_with({hub_logs=true}, "NOTIFICATION[DEBUG] TAMPER CLEAR!")
+        contact_event = capabilities.tamperAlert.tamper.clear()
+      else
+        log.info_with({hub_logs=true},
+                      string.format("NOTIFICATION[DEBUG] UNHANDLED SYSTEM CLEAR EVENT:%d", 
+                      notificationEvent))
+      end
+
+    elseif (notificationType == Notification.notification_type.ACCESS_CONTROL) then
+
+      if (notificationEvent == 
+          Notification.event.access_control.BARRIER_SENSOR_LOW_BATTERY_WARNING) then
+        barrier_event = capabilities.doorControl.door.closed()
+        contact_event = capabilities.healthCheck.healthStatus.online()
+        contact_event = capabilities.battery.battery(CONTACTSENSOR_BATTERY_LEVEL_NORMAL)
```

Comment:
you don't seem to actually be emitting `barrier_event` or `contact_event` anywhere